modules/tensor_mechanics/src/materials/ADRadialReturnStressUpdate.C

```diff
@@ -56,6 +64,26 @@ ADRadialReturnStressUpdate::propagateQpStatefulPropertiesRadialReturn()
   _effective_inelastic_strain[_qp] = _effective_inelastic_strain_old[_qp];
 }
 
+int
+ADRadialReturnStressUpdate::calculateNumberSubsteps(const ADRankTwoTensor & strain_increment)
+{
+  unsigned int substep_number = 1;
+  // compute an effective elastic strain measure
+  const ADReal contracted_elastic_strain = strain_increment.doubleContraction(strain_increment);
```

Comment:
Thanks Daniel! This makes sense. So this could be:
`const Real contracted_elastic_strain = MetaPhysicL::raw_value(strain_increment).doubleContraction(MetaPhysicL::raw_value(strain_increment));`?